hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/SqlPageCodec.java

```diff
@@ -82,17 +92,55 @@ public static void encode(ClientMessage clientMessage, SqlPage sqlPage) {
 
                     break;
 
-                case SMALLINT:
-                case DECIMAL:
                 case REAL:
+                    ListCNFloatCodec.encode(clientMessage, (Iterable<Float>) column);
+
+                    break;
+
                 case DOUBLE:
+                    ListCNDoubleCodec.encode(clientMessage, (Iterable<Double>) column);
+
+                    break;
+
                 case DATE:
+                    ListCNLocalDateCodec.encode(clientMessage, (Iterable<LocalDate>) column);
+
+                    break;
+
                 case TIME:
+                    ListCNLocalTimeCodec.encode(clientMessage, (Iterable<LocalTime>) column);
+
+                    break;
+
                 case TIMESTAMP:
+                    ListCNLocalDateTimeCodec.encode(clientMessage, (Iterable<LocalDateTime>) column);
+
+                    break;
+
                 case TIMESTAMP_WITH_TIME_ZONE:
+                    ListCNOffsetDateTimeCodec.encode(clientMessage, (Iterable<OffsetDateTime>) column);
+
+                    break;
+
+                case DECIMAL:
+                    ListMultiFrameCodec.encode(clientMessage, (Iterable<BigDecimal>) column, BigDecimalCodec::encodeNullable);
+
+                    break;
+
                 case NULL:
+                    int size = 0;
```

Comment:
Minor: We could move this and decode functionality into something called `ListNullCodec` but I am fine with the current approach